crates/viewer/re_viewer/src/ui/top_panel.rs

```diff
@@ -426,103 +448,75 @@ fn memory_use_label_ui(ui: &mut egui::Ui, gpu_resource_stats: &WgpuResourcePoolS
     }
 
-    let text = format!("latency: {}", latency_text(e2e_latency_sec));
+    let text = format!("Latency: {}", latency_text(e2e));
     let response = ui.weak(text);
 
-    let hover_text = "End-to-end latency from when the data was logged by the SDK to when it is shown in the viewer.\n\
-                      This includes time for encoding, network latency, and decoding.\n\
-                      It is also affected by the framerate of the viewer.\n\
-                      This latency is inaccurate if the logging was done on a different machine, since it is clock-based.";
-
-    Some(response.on_hover_text(hover_text))
-}
-
-/// Shows the latency in the input queue.
-fn input_queue_latency_ui(ui: &mut egui::Ui, app: &mut App) {
-    let rx = app.msg_receive_set();
-
-    if rx.is_empty() {
-        return;
-    }
-
-    let is_latency_interesting = rx.sources().iter().any(|s| s.is_network());
-
-    let queue_len = rx.queue_len();
-
-    // empty queue == unreliable latency
-    let latency_sec = rx.latency_nanos() as f32 / 1e9;
-    if queue_len > 0 && (!is_latency_interesting || app.app_options().warn_latency < latency_sec) {
-        // we use this to avoid flicker
-        app.latest_queue_interest = web_time::Instant::now();
-    }
+    let e2e_hover_text = "End-to-end latency from when the data was logged by the SDK to when it is shown in the viewer.\n\
+    This includes time for encoding, network latency, and decoding.\n\
+    It is also affected by the framerate of the viewer.\n\
+    This latency is inaccurate if the logging was done on a different machine, since it is clock-based.";
+
+    let response = response.on_hover_ui(|ui| {
+        ui.ctx().request_repaint(); // The user is interested in the latency, so keep it updated.
+
+        egui::Grid::new("latency_snapshot")
+            .num_columns(2)
+            .striped(false)
+            .show(ui, |ui| {
+                ui.strong("log -> ingest (total end-to-end)")
+                    .on_hover_text(e2e_hover_text);
+                ui.monospace(latency_text(e2e));
+                ui.end_row();
+
+                ui.end_row();
+
+                ui.label("log -> chunk");
+                ui.monospace(latency_text(log2chunk));
+                ui.end_row();
+
+                if let Some(chunk2encode) = chunk2encode {
+                    ui.label("chunk -> encode");
+                    ui.monospace(latency_text(chunk2encode));
+                    ui.end_row();
+                }
+                if let Some(transmission) = transmission {
+                    ui.label("encode -> decode (transmission)");
+                    ui.monospace(latency_text(transmission));
+                    ui.end_row();
+                }
+                if let Some(decode2ingest) = decode2ingest {
+                    ui.label("decode -> ingest");
+                    ui.monospace(latency_text(decode2ingest));
+                    ui.end_row();
```

Comment:
Any suggestions on how we make this more understandable?

I was thinking maybe having it be like a chain instead:


```
  log
    ↓
  2.1ms
    ↓
batch
    ↓
  4.5ms
    ↓
SDK send
    ↓
  32.2ms
    ↓
Viewer receive
    ↓
  12.2ms
    ↓
 ingest
```
  
Would that make it clear that the measurement is the delay of the gap between each event?

Or should we name the gaps (e.g. "batching", "transmission", "input processing)?

Maybe @gavrelina has some thoughts

Reply:
I think that would be pretty nice the way you describe it. Main advantage with naming the gaps is brevity, but it makes it less clear how things are connected

But I trust @gavrelina's judgement more than my own here if she has more suggestions!

Reply:
![image](https://github.com/user-attachments/assets/f73a0b3a-365a-412b-aa7e-f6e4d3a95620)